File: 0001TwoSum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let hash = {};

  for(i = 0; i < nums.length; i++) {
    if(hash[target - nums[i]] !== undefined) {
      return [hash[target - nums[i]], i];
    }
    hash[nums[i]] = i;
  }
  return [];
};
// array solution 
// var twoSum = function(nums, target) {
//     for (x = 0; x < nums.length; x++) 
//     {
//         for (y = x + 1; y < nums.length; y++) 
//         {   
//             if ((nums[x] + nums[y]) == target) 
//             {
//                 var arr = [x, y]
//                 return arr;
//             }
//         }
//     }
//     return null
// };
//completed

File: Solutions/C++/0001TwoSum.cpp
class Solution
{
public:
    vector<int> twoSum(vector<int> &nums, int target)
    {
        std::unordered_map<int, int> hash;
        for (int i = 0; i < nums.size(); i++)
        {
            int half = target - nums[i];

            if (hash.find(half) != hash.end())
            {
                return {i, hash[half]};
            }

            hash[nums[i]] = i;
        }

        return {};
    }
};